tasks/VisitMeasureTasks.py

```diff
@@ -0,0 +1,15 @@
+import astropy.units as u
+from lsst.pipe.base import Struct, Task
+from lsst.pex.config import Config
+from lsst.verify import Measurement
+
+class NumSourcesTask(Task):
```

Comment:
It makes me pretty uncomfortable to duplicate code like this.  If we are going to have identical classes in two places, it should be factored out and imported in those two places rather than copying it.

Reply:
Thanks for flagging this issue. I have added a `GeneralMeasureTasks.py` to hold measurements that are shared between contexts. I modified both the matched catalog and single-visit analyses to use this shared measurement task for counting sources. Since these changes involved some implementation decisions and modifications to several files, I would appreciate if @SimonKrughoff could take a quick look before merging.